### PR TITLE
TS-4590: INKContInternal & INKVConnInternal Optimize

### PR DESCRIPTION
**Pull request by oknet**

### CODE REVIEW

File: proxy/InkAPI.cc [hunk 1]
```diff
@@ -1037,43 +1047,36 @@ INKVConnInternal::INKVConnInternal(TSEventFunc funcp, TSMutex mutexp)
   : INKContInternal(funcp, mutexp), m_read_vio(), m_write_vio(), m_output_vc(NULL)
 {
   m_closed = 0;
-  SET_HANDLER(&INKVConnInternal::handle_event);
 }
```
Comment (shinrich) [inline proxy/InkAPI.cc]: Why do we not sent the handler here?

File: proxy/InkAPI.cc [hunk 2]
```diff
@@ -1037,43 +1047,36 @@ INKVConnInternal::INKVConnInternal(TSEventFunc funcp, TSMutex mutexp)
...
-    INKVConnAllocator.free(this);
-  }
+  clear();
+  this->mutex.clear();
+  m_free_magic = INKCONT_INTERN_MAGIC_DEAD;
+  INKVConnAllocator.free(this);
 }
 
-int
-INKVConnInternal::handle_event(int event, void *edata)
```
Comment (shinrich) [inline proxy/InkAPI.cc]: Where did the INKVConnInternal::handle_event go?  Why is it not needed?  Do we not worry about destroying InkVConnInternal objects on the stack and leaving live references to hit as we unwind the stack?

**Comment by shinrich** [comment]
My main question is about the removal of handle_event.  The other changes to clean up look good.

**Comment by zwoop** [comment]
[approve ci]

File: proxy/InkAPI.cc (hunk 2 above)
Comment (oknet) [inline proxy/InkAPI.cc]: INKContInternal and INKVConnInternal share the one handle_event() and it is set in INKContInternal(funcp, mutexp).

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/264/ for details.

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/369/ for details.

**Comment by oknet** [comment]
@shinrich INKVConnInternal::handle_event is almost same as INKContInternal::handle_event but the INKContInternal::handle_event could handle period event callback status.

They are share handle_event_count() and handle_event_count() also accept INTERVAL event.

The INKVConnInternal::handle_event() should handle period event callback if handle_event_count() does. Then the INKVConnInternal::handle_event() is totally same as INKContInternal::handle_event().

**Comment by shinrich** [comment]
Ok, that makes sense.  Looks good then.  I can merge up later today.

**Pull request merged**